File: mobile_gara/lib/models/shift.dart
class ShiftModel {
  final int id;
  final String shiftName;
  final String startTime;
  final String endTime;
  final String description;
  final int maxMechanics;
  final int maxCashiers;

  ShiftModel({
    required this.id,
    required this.shiftName,
    required this.startTime,
    required this.endTime,
    required this.description,
    required this.maxMechanics,
    required this.maxCashiers,
  });

  factory ShiftModel.fromJson(Map<String, dynamic> json) {
    return ShiftModel(
      id: json['id'] ?? 0,
      shiftName: json['shiftName'] ?? '',
      startTime: json['startTime'] ?? '',
      endTime: json['endTime'] ?? '',
      description: json['description'] ?? '',
      maxMechanics: json['maxMechanics'] ?? 2,
      maxCashiers: json['maxCashiers'] ?? 1,
    );
  }
}

File: backend-services/auth-service/src/main/java/com/gara/auth_service/entity/Shift.java
package com.gara.auth_service.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalTime;

@Entity
@Data
@Table(name = "shifts")
@NoArgsConstructor
public class Shift {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "shift_name", nullable = false, length = 50)
    private String shiftName; // Ví dụ: Ca Sáng, Ca Chiều, Hành chính

    @Column(name = "start_time", nullable = false)
    private LocalTime startTime;

    @Column(name = "end_time", nullable = false)
    private LocalTime endTime;

    @Column(length = 255)
    private String description;

    @Column(name = "max_mechanics", nullable = false)
    private Integer maxMechanics = 2; // Mặc định 2 Thợ

    @Column(name = "max_cashiers", nullable = false)
    private Integer maxCashiers = 1; // Mặc định 1 Thu ngân
}
